src/ImmutableVirtualizedList/ImmutableVirtualizedList.js

```diff
@@ -81,6 +116,20 @@ class ImmutableVirtualizedList extends PureComponent {
   recordInteraction = (...args) =>
     this.virtualizedListRef && this.virtualizedListRef.recordInteraction(...args);
 
+  renderItem = (info) => {
+    const { renderSectionHeader, renderItem } = this.props;
+
+    if (renderSectionHeader) {
+      const renderMethod = this.state.stickyHeaderIndices.includes(info.item)
+        ? renderSectionHeader
+        : renderItem;
+
+      return renderMethod(info.item, this.keyExtractor(info.item, info.index));
```

Comment:
Ok the problem stems from this line. In your example app, you’re (understandably!) assuming that `renderItem()` will be called with the whole `info` object (as it is if we are dealing with a flat list without section headers, 3 lines below this comment). Will push a fix asap!